components/camel-whatsapp/src/main/java/org/apache/camel/component/whatsapp/service/WhatsAppServiceRestAPIJDKAdapter.java

```diff
@@ -0,0 +1,244 @@
+
+        try {
+            handler.sendMessage(exchange, callback, message, authorizationToken);
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeCamelException("Could not send message " + message, e);
+        }
+    }
+
+    static class OutgoingMediaMessageHandler
+            extends WhatsAppServiceRestAPIJDKAdapter.OutgoingMessageHandler<UploadMediaRequest> {
+
+        public OutgoingMediaMessageHandler(HttpClient httpClient, ObjectMapper mapper, String uri,
+                                           Class<? extends MessageResponse> resultClass) {
+            super(httpClient, mapper, uri, null, resultClass);
+        }
+
+        public OutgoingMediaMessageHandler(HttpClient httpClient, ObjectMapper mapper, String uri) {
+            this(httpClient, mapper, uri, MessageResponse.class);
+        }
+
+        @Override
+        protected void addBody(Builder builder, UploadMediaRequest message) {
+            Map<Object, Object> formData = new HashMap<>();
+            formData.put("messaging_product", "whatsapp");
+            formData.put("file", message.getUploadMedia());
+
+            String boundary = new BigInteger(256, new Random()).toString();
+            try {
+                builder.POST(ofMimeMultipartData(formData, boundary));
+            } catch (IOException e) {
+                throw new RuntimeCamelException("Could not serialize " + message, e);
+            }
+
+            builder.header("content-type", "multipart/form-data; boundary=" + boundary);
+        }
+
+        public static BodyPublisher ofMimeMultipartData(Map<Object, Object> data, String boundary) throws IOException {
+            var byteArrays = new ArrayList<byte[]>();
+            byte[] separator = ("--" + boundary + "\r\nContent-Disposition: form-data; name=").getBytes(StandardCharsets.UTF_8);
+            for (Map.Entry<Object, Object> entry : data.entrySet()) {
+                byteArrays.add(separator);
+
+                if (entry.getValue() instanceof UploadMedia) {
+                    UploadMedia uploadMedia = (UploadMedia) entry.getValue();
+                    byteArrays.add(("\"" + entry.getKey() + "\"; filename=\"" + uploadMedia.getFile().toPath().getFileName()
+                                    + "\"\r\nContent-Type: " + uploadMedia.getContentType()
+                                    + "\r\n\r\n").getBytes(StandardCharsets.UTF_8));
+                    byteArrays.add(Files.readAllBytes(uploadMedia.getFile().toPath()));
+                    byteArrays.add("\r\n".getBytes(StandardCharsets.UTF_8));
+                } else {
+                    byteArrays.add(("\"" + entry.getKey() + "\"\r\n\r\n" + entry.getValue() + "\r\n")
+                            .getBytes(StandardCharsets.UTF_8));
+                }
+            }
+            byteArrays.add(("--" + boundary + "--\r\n").getBytes(StandardCharsets.UTF_8));
+            return BodyPublishers.ofByteArrays(byteArrays);
```

Comment:
I'd avoid working directly with bytes here. Instead, I think you can make this code much simpler by using a `ByteBuffer` and then using it to generate the `byte[]` passed to the `BodyPublishers`.

Among other things, it can also help avoiding the long chains of string concatenation (which could be problematic if this part of the code is in the hot path).